### ISSUE TITLE
Bogus defaulting of identifier link

### ISSUE THREAD

**Issue by harpocrates** [haskell/haddock#1035]
Consider the following:

```haskell
module Swapped where

data Foo = Bar

data Bar = Foo

-- | A link to 'Bar'
foo :: ()
foo = ()
```

When I run the current `ghc-8.8` branch, I get the following console output

```
Haddock coverage:
Warning: Package name is not available.
Warning: 'Bar' is ambiguous. It is defined
    * at Swapped.hs:3:12
    * at Swapped.hs:5:1
    You may be able to disambiguate the identifier by qualifying it or
    by specifying the type/value namespace explicitly.
    Defaulting to the one defined at Swapped.hs:3:1
  25% (  1 /  4) in 'Swapped'
  Missing documentation for:
    Module header
    Foo (Swapped.hs:3)
    Bar (Swapped.hs:5)
```

Then the following bogus output (note that the `Bar` turns into a `Foo`!!!)

<img width="896" alt="screen shot 2019-02-26 at 8 31 30 am" src="https://user-images.githubusercontent.com/10766081/53429199-005b1d80-39a1-11e9-89b2-c2165f51f755.png">

/cc @sjakobi since I think you wrote the `ambiguous` function from which the problem seems to be coming...

**Comment by sjakobi** [comment]
Oops! Thanks for fixing this, Alec!

**Comment by harpocrates** [comment]
> Oops! Thanks for fixing this, Alec!

Does the fix look OK to you (I want to make sure I didn't inadvertently break something)?

**Comment by harpocrates** [comment]
Fixed in b682041ed1cbeaf5aa501f85e4e46a6d2e39da3a.

**Comment by sjakobi** [comment]
Here's another instance of this bug in the wild: http://hackage.haskell.org/package/witherable-0.3/docs/Data-Witherable.html#v:filter

A reference to `filter` turned into a reference to `Filterable`, the class that contains the `filter` method.

Thanks again for figuring it out and fixing it @harpocrates!

(cc @fumieval)
